test/metabase/test/util.clj

```diff
@@ -1186,3 +1186,9 @@
       (if (neg? @a)
         (apply f args)
         (throw (ex-info "Not yet" {:remaining @a}))))))
+
+(defmacro with-start-of-week
+  "Executes the body with `start-of-week` temporarily changed."
+  [start-of-week & body]
+  `(mt/with-temporary-setting-values [start-of-week ~start-of-week]
+     ~@body))
```

Comment:
Does this **really** need to be a macro, vs. just using `with-temporary-setting-values` directly? It doesn't save any lines of code... if you follow it to the logical extreme, why don't we have a separate `with-x` macro for every single Setting `x`? Because that would be crazy, right?

I think `with-temporary-setting-values` is actually a little clearer actually because it's clear you're changing a Setting as opposed to some system property or whatever.

Just my two cents